labcontrol/db/process.py

```diff
@@ -3819,3 +3826,407 @@ def _generate_amplicon_prep_information(self):
+                LEFT JOIN labman.well primersetwell2 ON (
+                    primersetcpcp2.container_id = primersetwell2.container_id)
+                LEFT JOIN labman.plate primersetplate ON (
+                    primersetwell.plate_id = primersetplate.plate_id)
+                LEFT JOIN labman.plate primersetplate2 ON (
+                    primersetwell2.plate_id = primersetplate2.plate_id)
+                FULL JOIN qiita.study_sample USING (sample_id)
+                LEFT JOIN qiita.study as study USING (study_id)
+                WHERE libprepplate.plate_id IN (
+                    SELECT distinct libprepplate2.plate_id
+                    FROM labman.sequencing_process sp
+                    LEFT JOIN labman.sequencing_process_lanes spl USING (
+                        sequencing_process_id)
+                    LEFT JOIN labman.pool_composition_components pcc1 ON (
+                        spl.pool_composition_id =
+                        pcc1.output_pool_composition_id)
+                   LEFT JOIN labman.library_prep_shotgun_composition libprepcp2
+                        ON (
+                        pcc1.input_composition_id = libprepcp2.composition_id)
+                    LEFT JOIN labman.composition libprepcpcp2 ON (
+                        libprepcp2.composition_id =
+                        libprepcpcp2.composition_id)
+                    LEFT JOIN labman.library_prep_shotgun_process libpreppr2
+                        ON (libprepcpcp2.upstream_process_id =
+                        libpreppr2.process_id)
+                    LEFT JOIN labman.well libprepwell2 ON (
+                        libprepcpcp2.container_id = libprepwell2.container_id)
+                    LEFT JOIN labman.plate libprepplate2 ON (
+                        libprepwell2.plate_id = libprepplate2.plate_id)
+                    WHERE sequencing_process_id = %s)
+                """
+
+        with sql_connection.TRN as TRN:
+            TRN.add(sql, [self.id])
+
+            results = [dict(r) for r in TRN.execute_fetchindex()]
+
+            for d in results:
+                # assert that study_id and orig_name are only None when both
+                # are None; making this a control entry and not a sample entry
+                # (based on empirical data). However, check for both.
+                if d['study_id'] is None and d['orig_name'] is None:
+                    # adding this field allows the user to alter 'study_id'
+                    # and/or 'orig_name' downstream, w/out losing important
+                    # metadata. Also clearly names this property.
+                    d['is_control'] = True
+                else:
+                    d['is_control'] = False
+
+                # convert time from datetime objects to strings
+                # no timezone information appears with %z; Assume timestamp
+                # w/out timezone type in Pg.
+                # therefore, assume timestamps are local to San Diego and are
```

Comment:
I don't think this comment is relevant anymore now that Process.get_date_format is being used.